containers/src/Kokkos_Vector.hpp

```diff
@@ -21,6 +21,17 @@
 #define KOKKOS_IMPL_PUBLIC_INCLUDE_NOTDEFINED_VECTOR
 #endif
 
+#if !defined(KOKKOS_ENABLE_DEPRECATED_CODE_4) || \
+    defined(KOKKOS_ENABLE_DEPRECATION_WARNINGS)
+namespace {
+[[deprecated("Deprecated <Kokkos_Vector.hpp> header is included")]] int
+emit_warning_kokkos_vector_deprecated() {
+  return 0;
+}
+static auto do_not_include = emit_warning_kokkos_vector_deprecated();
+}  // namespace
+#endif
```

Comment:
```suggestion
#include <Kokkos_Macros.hpp>

#if defined(KOKKOS_ENABLE_DEPRECATED_CODE_4)
namespace {
KOKKOS_DEPRECATED_WITH_COMMENT("Deprecated <Kokkos_Vector.hpp> header is included") int
emit_warning_kokkos_vector_deprecated() {
  return 0;
}
static auto do_not_include = emit_warning_kokkos_vector_deprecated();
}  // namespace
#else
#error "Deprecated <Kokkos_Vector.hpp> header is included"
#endif
```

The 
```
#include <Kokkos_Macros.hpp>
```
is crucial for fixing CI.

Reply:
Please go ahead and push to my branch